cli/cmd/add/component-version/cmd.go

```diff
@@ -224,15 +226,39 @@ func GetRepositorySpec(cmd *cobra.Command) (runtime.Typed, error) {
 	if err != nil {
 		return nil, fmt.Errorf("getting repository reference flag failed: %w", err)
 	}
-	var accessMode ctfv1.AccessMode = ctfv1.AccessModeReadWrite
-	if !repoRef.Exists() {
-		accessMode += "|" + ctfv1.AccessModeCreate
+
+	// Get the repository type.
+	repoTypeString, err := compref.GuessType(repoRef.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to guess repository type: %w", err)
+	}
+
+	rtyp, err := runtime.TypeFromString(repoTypeString)
+	if err != nil {
+		return nil, fmt.Errorf("unknown type %q: %w", repoTypeString, err)
+	}
+
+	// create the type of the detected type.
+	typed, err := compref.RepositoryScheme.NewObject(rtyp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create repository of type %q: %w", repoTypeString, err)
 	}
-	repoSpec := ctfv1.Repository{
-		Path:       repoRef.String(),
-		AccessMode: accessMode,
+
+	switch t := typed.(type) {
+	case *ociv1.Repository:
```

Comment:
I feel like this is a pretty hacky way to get back to the spec even though you could have gotten this directly from compref

Reply:
Talked about this. :D Will refactor things. :) Back to Draft.